### PR TITLE
feat: job update

### PR DESCRIPTION
**Pull request by Myles181**
# Update Job Route Implementation

## Description

This pull request introduces the implementation of the job update route in our application. The update route allows users to modify existing job entries with new data. The update functionality is essential for maintaining accurate and up-to-date job postings.

## Motivation and Context

This change is required to enable users to update existing job entries, ensuring that the job postings reflect the latest information. It solves the problem of outdated job postings, providing a way for users to manage and edit job details as necessary.

## How Has This Been Tested?

- **Unit Tests**: Added unit tests to verify the update route functionality, ensuring that jobs are correctly updated with valid data and appropriate error messages are returned for invalid requests.
- **Integration Tests**: Conducted integration tests using Postman to confirm that the update route interacts correctly with the database and handles various edge cases.
- **Manual Testing**: Performed manual testing in a development environment to ensure that the user interface for updating job postings works as expected.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/job_update.py
Comment (joboy-dev) [inline api/v1/routes/job_update.py:1]: I have said it over and over again not to make db queries in routes. There should be separation of concern
Check services folder on how to use services

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Rename your PR to feat: job update
Link this PR to an open issue. If there is none, create one
Write tests as well

File: api/v1/routes/job_update.py
Comment (Myles181) [reply api/v1/routes/job_update.py:1]: I just updated it Sir

File: api/v1/services/job.py
Comment (joboy-dev) [inline api/v1/services/job.py:1]: Alright, do something for me
Check other service files like the faq service to see how update was implemented in the service and update your code accordingly
Thanks

File: api/v1/routes/job_update.py
Comment (joboy-dev) [inline api/v1/routes/job_update.py:1]: Check faq routes file for how to handle update requests. Thanks

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes and I will merge. This is not bad

**Pull request closed without merging**